Factory/ClientFactory.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace BestItBugsnag\Factory;
+
+use Bugsnag\Client;
+use Bugsnag\Report;
+use RuntimeException;
+
+/**
+ * Factory to create a bugsnag client from a given config
+ *
+ * @package BestItBugsnag\Factory
+ */
+class ClientFactory
+{
+    /**
+     * The plugin config
+     *
+     * @var array
+     */
+    private $config;
+
+    /**
+     * The shop environment
+     *
+     * @var string
+     */
+    private $environment;
+
+    /**
+     * ClientFactory constructor.
+     *
+     * @param string $environment
+     * @param array $config
+     */
+    public function __construct(string $environment, array $config)
+    {
+        $this->environment = $environment;
+        $this->config = $config;
+    }
+
+    /**
+     * Creates the bugsnag client
+     *
+     * @return Client
+     */
+    public function create(): Client
+    {
+        if (empty($this->config['apiKey'])) {
```

Comment:
Is the client usable without a apiKey?

Reply:
Bugsnag by default tries to get it from the $_ENV variables. Thats what it does if no key is passed.